spec/system/ai_helper/ai_post_helper_spec.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require_relative "../../support/openai_completions_inference_stubs"
+
+RSpec.describe "AI Composer helper", type: :system, js: true do
+  fab!(:user) { Fabricate(:admin) }
+  fab!(:non_member_group) { Fabricate(:group) }
+  fab!(:topic) { Fabricate(:topic) }
+  fab!(:post) do
+    Fabricate(
+      :post,
+      topic: topic,
+      raw:
+        "I like to eat pie. It is a very good dessert. Some people are wasteful by throwing pie at others but I do not do that. I always eat the pie.",
+    )
+  end
+  fab!(:post_2) do
+    Fabricate(:post, topic: topic, raw: OpenAiCompletionsInferenceStubs.spanish_text)
+  end
+  let(:topic_page) { PageObjects::Pages::Topic.new }
+  let(:post_ai_helper) { PageObjects::Components::AIHelperPostOptions.new }
+
+  before do
+    Group.find_by(id: Group::AUTO_GROUPS[:admins]).add(user)
+    SiteSetting.composer_ai_helper_enabled = true
+    sign_in(user)
+  end
+
+  def select_post_text(selected_post)
+    topic_page.visit_topic(topic)
+    page.execute_script(
+      "var element = document.querySelector('#{topic_page.post_by_number_selector(selected_post.post_number)} .cooked p'); " +
+        "var range = document.createRange(); " + "range.selectNodeContents(element); " +
+        "var selection = window.getSelection(); " + "selection.removeAllRanges(); " +
+        "selection.addRange(range);",
+    )
+  end
+
+  context "when triggering AI helper in post" do
+    it "shows the Ask AI button in the post selection toolbar" do
+      select_post_text(post)
+      expect(post_ai_helper).to have_post_selection_toolbar
+      expect(post_ai_helper).to have_post_ai_helper
+    end
+
+    it "shows AI helper options after clicking the AI button" do
+      select_post_text(post)
+      post_ai_helper.click_ai_button
+      expect(post_ai_helper).to have_no_post_selection_primary_buttons
+      expect(post_ai_helper).to have_post_ai_helper_options
+    end
+
+    context "when using explain mode" do
+      skip "TODO: Fix explain mode option not appearing in spec" do
```

Comment:
Just a thought, no need to do anything, but maybe we could test these with a component or controller test?